tests/src/Integration/Action/SendEmailTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\SendEmailTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\SendEmail
+ * @group rules_actions
+ */
+class SendEmailTest extends RulesIntegrationTestBase {
+
+  protected $logger;
+  protected $mailManager;
+
+  /**
+   * The action to be tested.
+   *
+   * @var \Drupal\rules\Core\RulesActionInterface
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->logger = $this->getMock('Psr\Log\LoggerInterface');
+
+    $this->mailManager = $this->getMockBuilder('Drupal\Core\Mail\MailManagerInterface')
+      ->disableOriginalConstructor()
+      ->getMock();
+
+    $this->container->set('logger.factory', $this->logger);
+    $this->container->set('plugin.manager.mail', $this->mailManager);
+
+    $this->action = $this->actionManager->createInstance('rules_send_email');
+  }
+
+
+  /**
+   * Tests the summary.
+   *
+   * @covers ::summary
+   */
+  public function testSummary() {
+    $this->assertEquals('Send email', $this->action->summary());
+  }
+
+  /**
+   * Tests the action execution.
+   *
+   * @covers ::execute
+   */
+  public function testActionExecution() {
+    $params = array(
+      'to' => "mail@example.com",
+      'from' => "admin@example.com",
+      'result' => TRUE,
+    );
+
+    $this->mailManager
+      ->expects($this->any())
+      ->method('mail')
```

Comment:
->with() is missing here where you specify the expected parameters to mail() method call.